src/main/scala/com/johnsnowlabs/nlp/annotators/pos/perceptron/PerceptronApproach.scala

```diff
@@ -89,26 +76,26 @@ class PerceptronApproach(override val uid: String) extends AnnotatorApproach[Per
     /**
       * Generates TagBook, which holds all the word to tags mapping that are not ambiguous
       */
-    val taggedSentences: Array[TaggedSentence] = if (get(posCol).isDefined) {
+    val taggedSentences: Array[TaggedSentence] = {
       import ResourceHelper.spark.implicits._
-      val tokenColumn = dataset.schema.fields
-        .find(f => f.metadata.contains("annotatorType") && f.metadata.getString("annotatorType") == AnnotatorType.TOKEN)
-        .map(_.name).get
-      dataset.select(tokenColumn, $(posCol))
-        .as[(Array[Annotation], Array[String])]
+
+      val datasetSchemaFields = dataset.schema.fields
+        .find(f => f.metadata.contains("annotatorType") && f.metadata.getString("annotatorType") == AnnotatorType.POS)
+
+      require(datasetSchemaFields.map(_.name).isDefined, s"Cannot train from DataFrame without POS annotatorType by posCol")
+
+      val posColumn = datasetSchemaFields.map(_.name).get
+
```

Comment:
Small note here:
```
val datasetSchemaFields = dataset.schema.fields
        .find(f => f.metadata.contains("annotatorType") && f.metadata.getString("annotatorType") == AnnotatorType.POS
```
This will result either in the name of column with `AnnotatorType.POS` or in `None`.  The reason I seprate the `map(_.name).get` and put it after `require` was to make sure it doesn't crash in case there is no `POS` column without telling user why it crashed. 
The `require` makes sure user knows there is no `POS` clolumn instead of crashing without explaination.